samples/set_best_video_mode/main.cpp

```diff
@@ -0,0 +1,70 @@
+#include <hal/debug.h>
+#include <hal/video.h>
+#include <windows.h>
+#include <vector>
+
+// Screen dimension globals
+static int SCREEN_WIDTH = 640;
+static int SCREEN_HEIGHT = 480;
+
+int main(void)
+{
+    // set the default video mode
+    XVideoSetMode(SCREEN_WIDTH, SCREEN_HEIGHT, 32, REFRESH_DEFAULT);
+
+    // create an empty list for the available video modes
+    std::vector<VIDEO_MODE> availableVideoModes;
+
+    // save the best video mode here
+    VIDEO_MODE bestVideoMode = {0, 0, 0, 0};
```

Comment:
* "best" is subjective. Best in what way? "Highest quality" seems like goal here.
* This feels like it should be a library function so applications don't have to reproduce this mode selection logic

Reply:
> Resolution is the primary factor, but it also accounts for bpp and refresh rate.

We can consider bpp (color resolution) and refresh rate (temporal resolution) part of "highest resolution" selection

Reply:
Right, I was just using 1080p as an example. I know that resolution is not support by og Xbox.